src/Serialization/Rings.jl

```diff
@@ -0,0 +1,53 @@
+################################################################################
+# fmpq polynomial ring
+function save_internal(s::SerializerState, R::FmpqMPolyRing)
+    base_ring = R.base_ring
+    return Dict(
+        :symbols => save_type_dispatch(s, [String(x) for x in R.S]),
+        :base_ring => save_type_dispatch(s, base_ring),
+        :nvars => save_type_dispatch(s, R.nvars),
+        :nfields => save_type_dispatch(s, R.nfields),
+    )
+end
+
+function load_internal(s::DeserializerState, ::Type{FmpqMPolyRing}, dict::Dict)
+    base_ring = eval(Meta.parse(dict[:base_ring][:type] * "()"))
```

Comment:
Should we really work with `eval` here? That seems awfully brittle (and highly inefficient -- the Julia parser is slow; although these are only tiny fragments we are parsing).

Isn't this what `encodeType` and `decodeType` are for? I.e. something like this (completely untested):
```suggestion
    ring_type = decodeType(dict[:base_ring][:type])
    base_ring = ring_type()
```
Although of course in this case this is probably useless because you could just hardcode
```suggestion
    base_ring = QQ
```
but I assume this is meant to show case how arbitrary `MPolyRing` subtypes could be dealt with?

Reply:
sorry, yes i I hadn't really notice the decodeType/encodeType functions,  so thanks I will change this.  I was thinking about arbitrary subtypes while writing this.